design/ceph/object/object-store-backend.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Object Store Backend
+target-version: release-1.10
+---
+
+# Overview
+
+Ceph RADOS Gateway is undergoing a major overhaul with the Zipper project. The goal of Project
+Zipper is to create an API that can be used to separate the RGW top end (represented by Swift and
+S3) from bottom end providers (represented currently by RADOS, and eventually by Cloud, File, and
+Memory backends). This should allow the front ends to access any implemented back end
+transparently.
+
+The secondary objective is to allow multiple layers of these API providers to stack up, allowing
+transformations of operations (such as redirection to different backends), or caching layers that
+cut across multiple backends.
+
+The primary Store is the current RadosStore, which provides access to a Ceph cluster via RADOS.
+However, new Stores are possible that store the data in any desired platform. One such Store, called
+DBStore, has been developed that stores data in SQL, and specifically in a local SQLite database.
+
+## Goals
+
+Rook-Ceph has a tremendous support for the Ceph Object Store and we want to leverage that.
+Especially the various controllers that are reconciling ObjectStore, Multisite configurations and
+Bucket provisioning.
+Mainly we are looking at providing the next generation of RGW via Rook.
+
+# Proposal details
+
+The `CephObjectStore` API will have a few details added to its spec such as an image so that it can
+be deployed without a Ceph cluster.
+
+```go
+type ObjectStoreSpec struct {
+...
+	// Image is the name of the image to use to deploy the object store
+	Image string `json:"image"`
```

Comment:
If it is still connected to the ceph project, I think it still makes sense to include it in the rook-ceph operator, even if the new CRD is not only for ceph or doesn't have ceph in the CRD name. Even today rook installs the OBC CRDs without the ceph name: `objectbuckets.objectbucket.io`

Reply:
I am wondering what does this image represents, IMO RGW will still package with ceph. For DB store backend we may need to include SQLite package or other interested backend is motr by Seagate(from cephalcon schedule) and may be suse has some interest with ranger. I can understand Rook can add support for generic backends like dbstore & not sure about the vendor-specific backends

Reply:
It's just the ceph image. Today the ObjectStore CRD knows which Ceph image to use because it's the same as the CephCluster one. If we don't have a CephCluster (rgw using db store) then the controller has no idea which Ceph image to use. Does that help?

Reply:
Ideally, I'd like to "extract"/"convert" CephObjectStore to ObjectStore and Ceph being one of the backends. Let's discuss more.